foreman/data_refinery_foreman/surveyor/harmony.py

```diff
@@ -0,0 +1,699 @@
+
+    ##
+    # Subject
+    ##
+    subject_fields = [
+                    # AE
+                    'subject', 
+                    'subject id', 
+                    'subject/sample source id', 
+                    'subject identifier', 
+                    'human subject anonymized id', 
+                    'individual', 
+                    'individual identifier', 
+                    'individual id', 
+                    'patient', 
+                    'patient id', 
+                    'patient identifier', 
+                    'patient number', 
+                    'patient no', 
+                    'donor id', 
+                    'donor',
+
+                    # SRA
+                    'sample_source_name'
+                ]
+    subject_fields = add_variants(subject_fields)
+    for sample in original_samples:
+        title = sample['title']
+        for key, value in sample.copy().items():
+            lower_key = key.lower().strip() 
+            if lower_key in subject_fields:
+                harmonized_samples[title]['subject'] = value.lower().strip()
+
+    ##
+    # Developement Stage!
+    ##
+    development_stage_fields = [
+                    'developmental stage', 
+                    'development stage', 
+                    'development stages'
+                ]
+    development_stage_fields = add_variants(development_stage_fields)
+    for sample in original_samples:
+        title = sample['title']
+        for key, value in sample.copy().items():
+            lower_key = key.lower().strip() 
+            if lower_key in development_stage_fields:
+                harmonized_samples[title]['developmental_stage'] = value.lower().strip()
+
+    ##
+    # Compound!
+    ##
+    compound_fields = [
+                    'compound', 
+                    'compound1', 
+                    'compound2', 
+                    'compound name', 
+                    'drug', 
+                    'drugs',
+                    'immunosuppressive drugs'
+                ]
+    compound_fields = add_variants(compound_fields)
+    for sample in original_samples:
+        title = sample['title']
+        for key, value in sample.copy().items():
+            lower_key = key.lower().strip() 
+            if lower_key in compound_fields:
+                harmonized_samples[title]['compound'] = value.lower().strip()
+
+    ##
+    # Time!
+    ##
+    time_fields = [
+                    'time', 
+                    'initial time point', 
+                    'start time', 
+                    'stop time', 
+                    'time point', 
+                    'sampling time point', 
+                    'sampling time',
+                    'time post infection'
+                ]
+    time_fields = add_variants(time_fields)
+    for sample in original_samples:
+        title = sample['title']
+        for key, value in sample.copy().items():
```

Comment:
This looks like a great method for dealing with the messiness of this metadata! Good work. The only comment I have on the `harmonize` function is about the lines like `for key, value in sample.copy().items():`. Why are you `copy()`ing every sample? The [view object](https://docs.python.org/3.6/library/stdtypes.html#dict-views) returned by `items()` doesn't allow you to mutate the original dictionary.